app/models/story.rb

```diff
@@ -61,4 +61,21 @@ def as_json(options = {})
   def readonly?
     !accepted_at_changed? && accepted_at.present?
   end
+
+  # Set the project start date to today if the project start date is nil
+  # and the state is changing to any state other than 'unstarted' or 'unscheduled'
+  def fix_project_start_date
+    return unless self.state_changed?
+    if self.project && !self.project.start_date && !['unstarted', 'unscheduled'].include?(self.state)
```

Comment:
Another minor note, you could get rid of most `self` references in here and on the method above as well.